src/components/app.jsx

```diff
@@ -1,9 +1,136 @@
-import React from "react";
-import { ReactComponent as Logo } from "../images/sun.svg";
-import "../App.css";
+import React, { useState } from 'react';
+import { getCoords, fetchWeather } from '../utils';
+import Input from './input';
+import Forecast from './forecast';
+import Button from './button';
+import cities from './data/cities';
+
+import dayjs from 'dayjs';
+
+import '../App.css';
 
 function App() {
-  return <h1>Weather</h1>;
+	const [show_weather, setShowWeather] = useState(false);
+	const [current_city, setCurrentCity] = useState('');
+	const [current_unit, setUnit] = useState('imperial');
+	const [forecast, setForecast] = useState([]);
+	const [current_weather, setCurrentWeather] = useState({
+		feels_like: '',
+		condition: '',
+		icon: '',
+		temp: '',
+		temp_max: '',
+		temp_min: '',
+	});
+	const [coords, setCoords] = useState({
+		lat: '',
+		lon: '',
+	});
+
+	const onInput = city => {
+		setShowWeather(true);
+		setCurrentCity(city);
+
+		getCoords(city)
+			.then(res => {
+				const coords = res;
+				setCoords(coords);
+
+				setWeatherData(city, current_unit, coords);
+			});
+	};
+
+	const toggleUnit = () => {
+		const unit = current_unit === 'imperial' ? 'metric' : 'imperial';
+		setUnit(unit);
+
+		setWeatherData(current_city, unit, coords);
```

Comment:
It looks like you're doing a brand new HTTP request each time the units are toggled. If you were asked to make toggling units more efficient, what are some ways you might try to do that?

Reply:
Yeah, this was one part that I kept thinking back on during and even after the PR.. definitely not efficient to request twice! But thinking about it now, I would probably consider writing a function that converts the units from imperial to metric somewhere in the fetch method and include that in the response and structure the response back to something like this : 

`{
	metric: {
		feels_like: '',
		condition: '',
		icon: '',
		temp: '',
		temp_max: '',
		temp_min: '',
	},
	imperial: {
		feels_like: '',
		condition: '',
		icon: '',
		temp: '',
		temp_max: '',
		temp_min: '',
	}
}`

That way, we can easily switch between the two units throughout the app.